code/Protein.py

```diff
@@ -49,22 +55,57 @@ def print_information(self):
 		print('	annotations:', self.annotations)
 		      
 	def standardize(self, means, std_devs, projection_matrix) -> np.array:
-                """Standadize dataset:
+				"""Standadize dataset:
                 param: means: np.array
                 param: std_devs: np.array
                 param: projection_matrix: np.array
                 output: reduced_dataset"""
-                dataset = self.model_raw_data
-                std_dataset = np.zeros(dataset.shape)
-                std_dataset = (dataset - means) / std_devs
-                reduced_dataset = std_dataset.dot(projection_matrix)
-                return reduced_dataset
+				dataset = self.model_raw_data
+				std_dataset = np.zeros(dataset.shape)
+				std_dataset = (dataset - means) / std_devs
+				reduced_dataset = std_dataset.dot(projection_matrix)
+				return reduced_dataset
+	def provide_raw_loops(self, transmem_doms_limit):
+    
+		if transmem_doms_limit == 0:
+        
+        
+			new_seq = ''
+			i_lengths = []
+			for t, label in zip(self.sequence, self.tmhmm_seq):
+				if label in ['o', 'O']:
+					new_seq += t
+				elif label in ['m', 'M']:
+					new_seq += 'X'
+				elif label in ['i', 'I']:
+					new_seq += ''
+				i_lengths.append(1)
+                
+			avg_i_length = sum(i_lengths) / len(i_lengths) if i_lengths else 0
+      
+        
+			new_seq = new_seq.replace('X', '')
+        
+            
+			return new_seq
+		else:
+			conds = ['o', 'O']
+			if self.localization == "out":
+				conds += ['i', 'I']
+			new_seq = ""
+			for i in range(self.length):
+				if self.tmhmm_seq[i] in conds:
+					new_seq += self.sequence[i]
+				elif len(new_seq) > 0 and not new_seq[len(new_seq)-1] == "X":
+					new_seq += "X"
+			return new_seq.split('X')
+
 	
-	def provide_raw_loops(self):
+	def provide_raw_loops_std(self):
```

Comment:
Anche qui non mi e' chiaro

Reply:
qui ho definito una nuova funziona razor che si attiva solo se transmem_doms_limit viene impostato a zero, ovvero se si vuole recuperare da tutte le proteine transmem solo le parti outer. l'ho fatto perchè nel caso di transmem>2 razor opera in modo classico ma nella predizione degli epitopi verrebbero incluse anche porzioni interne